### PR TITLE
Valhal auth merge

### PR DESCRIPTION
**Pull request by davidgrove73**
removed old authority code from valhal
added the new authority gem
updated letter_book edit form - added authorities

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -22,11 +22,14 @@
...
-//= require viaf/jquery.viaf.autocomplete.js
-//= require viaf/viaf_util.js
+// Required by Authority gem
+//= require authority/viaf/jquery.viafauto.js
+//= require authority/viaf/jquery.viafauto.extended.js
+//= require authority/viaf/viaf_util.js
+// require authority/viaf/viafImport.js
+//= require authority/typeahead.bundle.js
+//= require authority/authority_typeahead.js
+
```
Comment (davidgrove73) [inline app/assets/javascripts/application.js]: This should be the right order

File: app/assets/javascripts/work.js.erb [hunk 2]
```diff
@@ -321,122 +331,44 @@ function renderImageFoundModal(event) {
...
-                            modal_agent_field = event.target.id;
-                            console.log("modal_agent_field "+modal_agent_field);
-                            viafImport();
-                        }
-                    });
-                }
-            });
 
+// modal functionallity
+function modalSaveForm(form,content_div) {
```
Comment (davidgrove73) [inline app/assets/javascripts/work.js.erb]: This is used to submit new person/org  forms inside a modal by ajax request

File: app/controllers/letter_books_controller.rb [hunk 3]
```diff
@@ -29,11 +29,26 @@ def set_letter_book
   end
 
   def work_params
-    params[:letter_book].permit( :origin_date, titles_attributes: [[:id, :value, :subtitle, :lang, :type]])
+    params[:letter_book].permit(:language, :origin_date, titles_attributes: [[:id, :value, :subtitle, :lang, :type, :_destroy]],
```
Comment (davidgrove73) [inline app/controllers/letter_books_controller.rb]: make sure all params are acceptet for letter_books

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -24,10 +24,12 @@ def dissemination_profiles
   # displayed is either the element's label if present
   # or the element's name if not.
   # The list is sorted by the value of the labels ascending
-  def get_list_with_labels(list_name)
+  # if a recstrict_to argument is given, array is restricted to elements with the name in the list
+  def get_list_with_labels(list_name,restrict_to=[])
     list = Administration::ControlledList.with(:name, list_name)
     elements = list.nil? ? [] : list.elements.to_a
     elements.map!{ |e| [ (e.label.present? ? e.label : e.name), e.name] }
+    elements.select!{|e| restrict_to.include?(e.last)} if restrict_to.present?
```
Comment (davidgrove73) [inline app/helpers/application_helper.rb]: enables the function to give only a subset of values in controlled list

File: app/models/datastreams/trans_walker.rb [hunk 5]
```diff
@@ -41,11 +41,9 @@ def to_work(mods)
         date   = p.xpath('mods:namePart[@type="date"]',
                          'mods'=>ns).text
 
-        if family.present? || given.present?
-          author = Authority::Person.new(given_name: given, family_name: family)
-        else
-          author = Authority::Person.new(_name: full)
-        end
+
+        author = Authority::Person.find_or_create(given_name: given, family_name: family)
```
Comment (davidgrove73) [inline app/models/datastreams/trans_walker.rb]: So that Tove Ditlevsen is only create once

File: app/views/layouts/modal.html.erb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
```
Comment (davidgrove73) [inline app/views/layouts/modal.html.erb]: This modal renders a new person/org form inside a bootstrap modal.
If a param 'modal=true' is set in the request to the Authority Gem

Check also the code in the top of there person/organization controllers in the authority Gem

**Comment by xenma** [comment]
I think I got an overview, but I definitely couldn't do a code review :D 
:+1:

**Pull request merged**